### ISSUE TITLE
Misleading error when creating a file in a  rozofs mountpoint via root

### ISSUE THREAD

**Issue by lbeltrame** [rozofs/rozofs#154]
Found this one almost by mistake. Example:

```
$ touch /rozomount/test.txt
$
```

But as root:

```
# touch /rozomount/test.txt
touch: cannot touch ‘/rozomount/test.txt’: No space left on device
```

This *only* occurs for file creation. If the file is existing, it can be accessed and changed properly.

**Comment by rozodopamine** [comment]
Luca,
your problem probably comes from an enhancement introduced in 2.1.13 that
controls the meta-data resources and
refuses file creation with ENOSPACE when these resources have gone under a
given threshold.. This prevent from
creating inconsistent meta data that may be difficult to cleanup manually
later.

You can check the meta-data resources on your active export node with
rozodiag -T export:1 -c meta

You can adjust the thresholds to your system if the default ones are to
small by setting
min_metadata_inodes as well as min_metadata_MB in /etc/rozofs/rozofs.conf.
Also check man rozofs.conf.
Regards,
JPierre




Jean-Pierre Monchanin - Customer support


*jean-pierre.monchanin@rozofs.com
<jean-pierre.monchanin@rozofs.com>www.rozofs.com
<http://www.rozofs.com>https://github.com/rozofs/rozofs
<https://github.com/rozofs/rozofs>*



2017-07-11 16:33 GMT+02:00 Luca Beltrame <notifications@github.com>:

**Comment by lbeltrame** [comment]
It doesn't look I'm close to the metadata limit:

# rozodiag -T export:1 -c meta
____[exportd-S1 ]__[meta ]____
{ "meta-data" : {
     "eid"             : 1,
     "full"            : "NO",
     "full counter"    : 0,
     "fstat errors"    : 0,
     "inodes" : {
        "total"     : 366264320,
        "free"      : 365285870,
        "mini"      : 262144,
        "depletion" : "NO"
     },
     "sizeMB" : {
        "total"     : 22800216,
        "free"      : 21281533,
        "mini"      : 2048,
        "depletion" : "NO"
     }
  }
}

I just double checked on one affected node and it still gives me ENOSPACE if I try as root. Not a big deal but still a bit counter-intuitive.

**Comment by rozodopamine** [comment]
You have no had any meta data space depletion, so
rozodiag -T export:1 -c all
could probably help.
Regards,
JPierre

Jean-Pierre Monchanin - Customer support


*jean-pierre.monchanin@rozofs.com
<jean-pierre.monchanin@rozofs.com>www.rozofs.com
<http://www.rozofs.com>https://github.com/rozofs/rozofs
<https://github.com/rozofs/rozofs>*



2017-07-12 16:19 GMT+02:00 Luca Beltrame <notifications@github.com>:

**Comment by lbeltrame** [comment]
Here's the output: https://gist.github.com/lbeltrame/343020fbf16bc5718747dd3fd716fbc0

**Comment by lbeltrame** [comment]
Forgot to say, this is on the export host.

**Comment by rozodopamine** [comment]
Could provide us with the rozofsmount trace to figure exactly what is
wrong..
Activate the trace this way:
rozodiag -T mount:0 -c trc enable -c trc count 400

Then run the test and finally dump the trace
rozodiag -T mount:0 -c trc

Regards,
JPierre

Jean-Pierre Monchanin - Customer support


*jean-pierre.monchanin@rozofs.com
<jean-pierre.monchanin@rozofs.com>www.rozofs.com
<http://www.rozofs.com>https://github.com/rozofs/rozofs
<https://github.com/rozofs/rozofs>*



2017-07-12 16:32 GMT+02:00 Luca Beltrame <notifications@github.com>:

**Comment by lbeltrame** [comment]
It's surprisingly short (as root):

```
____[rozofsmount 0]__[trc ]____
trace entry size : 152 Bytes
ino size         : 8 Bytes
[           0 ]--> create      0 188600000000007a test (0x8841)
[     1011182 ]<-- create      0 000000000000 00000000-0000-0000-0000-000000000000 28:No space left on device
```

OTOH, a working (as user) trace is:

```
____[rozofsmount 0]__[trc ]____
trace entry size : 152 Bytes
ino size         : 8 Bytes
[           0 ]--> getattr     0 188600000000007a 00000000-0000-0000-0000-000000000000
[      974474 ]<-- getattr     0 188600000000007a 00000000-0000-4000-7a00-000000008618 0:Success     4096
[      120008 ]--> lookup      1 188600000000007a test (0x0)
[      595706 ]<-- lookup      1 000000000000 00000000-0000-0000-0000-000000000000 2:No such file or directory 18446744073709551615
[       44545 ]--> create      2 188600000000007a test (0x8841)
[     1026614 ]<-- create      2 11e9000000000c18 00000000-0000-4000-180c-00000000e911 0:Success
[      138582 ]--> flush       3 7fb0b4d13de0 00000000-0000-4000-180c-00000000e911        0/0
[        8640 ]<-- flush       3 7fb0b4d13de0 00000000-0000-4000-180c-00000000e911 0:Success
[      124612 ]--> setattr     4 11e9000000000c18 00000000-0000-0000-0000-000000000000  ATIME MTIME
[      757671 ]<-- setattr     4 11e9000000000c18 00000000-0000-4000-180c-00000000e911 0:Success        0
[       25745 ]--> removexattr    5 11e9000000000c18 security.ima
[      466272 ]<-- removexattr    5 11e9000000000c18 00000000-0000-0000-0000-000000000000 61:No data available
[       24185 ]--> flush       6 7fb0b4d13de0 00000000-0000-4000-180c-00000000e911        0/0
[        3184 ]<-- flush       6 7fb0b4d13de0 00000000-0000-4000-180c-00000000e911 0:Success
[       66915 ]--> release     7 7fb0b4d13de0 00000000-0000-0000-0000-000000000000
[       71789 ]<-- release     7 7fb0b4d13de0 00000000-0000-4000-180c-00000000e911 0:Success
[     9869759 ]--> getattr     8 188600000000007a 00000000-0000-0000-0000-000000000000
[       30380 ]<-- getattr     8 000000000000 00000000-0000-4000-7a00-000000008618 0:Success     4096
```

**Comment by rozodopamine** [comment]
Luca,
can you run "id root"
Thanks,
JPierre

Jean-Pierre Monchanin - Customer support


*jean-pierre.monchanin@rozofs.com
<jean-pierre.monchanin@rozofs.com>www.rozofs.com
<http://www.rozofs.com>https://github.com/rozofs/rozofs
<https://github.com/rozofs/rozofs>*



2017-07-13 9:38 GMT+02:00 Luca Beltrame <notifications@github.com>:

**Comment by lbeltrame** [comment]
Sorry, somehow this slipped off my radar:

```
id root
uid=0(root) gid=0(root) groups=0(root)
```

With regards to users, we're using LDAP to handle user accounts.

**Comment by rozonicotine** [comment]
Fixed since release 2.1.42
